### PR TITLE
Create LSP14Ownable2Step specs 

### PR DESCRIPTION
**Pull request by b00ste**
## What does this PR introduce?

This PR introduces LSP14Ownable2Step, an LSP focused towards the following processes:
- owning a contract
- transferring ownership of a contract
- renouncing ownership of a contract

Introduced features:
- Transfer ownership in 2 steps.
- Renounce ownership in 2 steps.
- Notify the new owner whenever the first step of transfer ownership is executed.
- Notify both the old owner and the new owner whenever the process of transferring ownership is complete.

### CODE REVIEW

File: LSPs/LSP-0-ERC725Account.md [hunk 1]
```diff
@@ -77,71 +77,15 @@ Contains the methods from:
...
+- [LSP14](./LSP-14-Ownable2Step.md#specification)
 
 #### transferOwnership
 
-```solidity
-function transferOwnership(address newOwner) external;
-```
-
-Sets the `newOwner` as `pendingOwner`.
+Similar to [`transferOwnership(address)`](./LSP-14-Ownable2Step.md#transferownership) from LSP14.
```
Comment (YamenMerhi) [inline LSPs/LSP-0-ERC725Account.md:84]: I would suggest doing it this way ⬇️ What do you think @CJ42 @b00ste? 

```solidity
function transferOwnership(address newOwner) external;
```
Similar to [`transferOwnership(address)`](./LSP-14-Ownable2Step.md#transferownership) from LSP14.

**Additional requirements:**

- The `newOwner` MUST NOT be the contract itself `address(this)`.

File: LSPs/LSP-9-Vault.md [hunk 2]
```diff
@@ -71,6 +71,14 @@ See the [Interface Cheat Sheet](#interface-cheat-sheet) for details.
 
 **Contains the methods from** [ERC725](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-725.md#specification) (General data key-value store, and general executor) with overriding:
 
+#### transferOwnership
```
Comment (YamenMerhi) [inline LSPs/LSP-9-Vault.md:74]: It should be re-arrenged. The block is inserted in the wrong place.
It should be:
**Contains the methods from:** 
- [ERC725] with overriding:

#### execute

```solidity
function execute(uint256 operationType, address to, uint256 value, bytes memory data) public payable returns(bytes memory)
```
[...]

- [LSP1]
- [LSP14] with overriding: 

#### transferOwnership

```solidity
function transferOwnership(address newOwner) external;
```
Similar to [`transferOwnership(address)`](./LSP-14-Ownable2Step.md#transferownership) from LSP14.

**Additional requirements:**

- The `newOwner` MUST NOT be the contract itself `address(this)`.

File: LSPs/LSP-14-Ownable2Step.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+- notify when the new owner of the contract should accept ownership.
+- notify the previous and new owner when ownership of the contract has been fully transferred.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+Because owning the contract allows access to sensitive methods, transferring to the wrong address or renouncing ownership of the contract by accident in a single transaction can be highly dangerous. Having those two processes work in 2 steps substantially reduces the probability of transferring or renouncing ownership of the contract by accident.
+
+## Motivation
+<!--The motivation is critical for LIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the LIP solves. LIP submissions without sufficient motivation may be rejected outright.-->
+The particular issue that this implementation of [EIP173](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-173.md) solves is the irreversible nature of transferring or renouncing ownership of a contract.
```
Comment (YamenMerhi) [inline LSPs/LSP-14-Ownable2Step.md:26]: > this implementation of [EIP173]

The word `implementation` seems weird to me

File: LSPs/LSP-9-Vault.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ discussions-to: https://discord.gg/E2rJPP4
 status: Draft
 type: LSP
 created: 2021-09-21
-requires: LSP1, LSP2, ERC165, ERC173, ERC725X, ERC725Y
+requires: LSP1, LSP2, LSP14, ERC165, ERC725X, ERC725Y
```
Comment (YamenMerhi) [inline LSPs/LSP-9-Vault.md:9]: I know this is old but can you re-arrange them so it starts with ERC165 --> ERC725X -- ERC725Y --> LSP1 etc. Thanks 🙏

File: LSPs/LSP-0-ERC725Account.md (hunk 1 above)
Comment (YamenMerhi) [reply LSPs/LSP-0-ERC725Account.md:84]: Also the same comment that i did to LSP9 Vault can be applied here.

Contains method from:
- ER725
- LSP1
- LSP14 with overriding: 

```solidity
function transferOwnership(address newOwner) external;
```
Similar to [`transferOwnership(address)`](./LSP-14-Ownable2Step.md#transferownership) from LSP14.

**Additional requirements:**

- The `newOwner` MUST NOT be the contract itself `address(this)`.

Comment (b00ste) [reply LSPs/LSP-0-ERC725Account.md:84]: Yeah, it looks good to me

File: LSPs/LSP-14-Ownable2Step.md (hunk 3 above)
Comment (b00ste) [reply LSPs/LSP-14-Ownable2Step.md:26]: " The particular issue that is present in [EIP173] " What about this?

File: LSPs/LSP-0-ERC725Account.md [hunk 5]
```diff
@@ -77,71 +77,19 @@ Contains the methods from:
...
+- [LSP14](./LSP-14-Ownable2Step.md#specification)
 
 #### transferOwnership
 
 ```solidity
 function transferOwnership(address newOwner) external;
 ```
 
-Sets the `newOwner` as `pendingOwner`.
+Similar to [`transferOwnership(address)`](./LSP-14-Ownable2Step.md#transferownership) from LSP14.
```
Comment (frozeman) [inline LSPs/LSP-0-ERC725Account.md:88]: This function is part of the LSP14 Specification, with additional requirements as follows:

File: LSPs/LSP-14-Ownable2Step.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+**Requirements:**
+
+- MUST only be called by the current `owner()` of the contract.
+
+**LSP1 Hooks:**
+
+- If the new owner address supports LSP1 interface, SHOULD call the new owner's [`universalReceiver(...)`] function with the following parameters below:
+
+    - `typeId`: keccak256('LSP14OwnershipTransferStarted')
```
Comment (frozeman) [inline LSPs/LSP-14-Ownable2Step.md:87]:
```
 - `typeId`: keccak256('LSP14OwnershipTransferStarted') : `0x123456789....`
```

**Review by frozeman — APPROVED** [review]

**Review by YamenMerhi — APPROVED** [review]

**Review by CJ42 — APPROVED** [review]

**Review by frozeman — APPROVED** [review]

**Pull request merged**